great_tables/_formats.py

```diff
@@ -2250,6 +2250,255 @@ def fmt_datetime_context(
+
+    Formatting with the `tf_style=` argument
+    ----------------------------------------
+    We need to supply a preset `tf_style=` value. The following table provides a listing of all
+    `tf_style=` values and their output `True` and `False` values.
+
+    |    | TF Style        | Output                  |
+    |----|-----------------|-------------------------|
+    | 1  | `"true-false"`  | `"true" / `"false"`     |
+    | 2  | `"yes-no"`      | `"yes" / `"no"`         |
+    | 3  | `"up-down"`     | `"up" / `"down"`        |
+    | 4  | `"check-mark"`  | `"✓" / `"✗"`            |
+    | 5  | `"circles"`     | `"●" / `"○"`            |
+    | 6  | `"squares"`     | `"■" / `"□"`            |
+    | 7  | `"diamonds"`    | `"◆" / `"◇"`            |
+    | 8  | `"arrows"`      | `"↑" / `"↓"`            |
+    | 9  | `"triangles"`   | `"▲" / `"▼"`            |
+    | 10 | `"triangles-lr"`| `"▶" / `"◀"`            |
+
+    Examples
+    --------
+    Let's use a subset of the `sp500` dataset to create a small table containing opening and closing
+    price data for the last few days in 2015. We added a boolean column (`dir`) where `True`
+    indicates a price increase from opening to closing and `False` is the opposite. Using `fmt_tf()`
+    generates up and down arrows in the `dir` column. We elect to use green upward arrows and red
+    downward arrows (through the `colors=` option).
+
+    ```{python}
+    from great_tables import GT
+    from great_tables.data import sp500
+    import polars as pl
+
+    sp500_mini = (
+        pl.from_pandas(sp500)
+        .slice(0, 5)
+        .drop(["volume", "adj_close", "high", "low"])
+        .with_columns(dir = pl.col("close") > pl.col("open"))
+    )
+
+    (
+        GT(sp500_mini, rowname_col="date")
+        .fmt_tf(columns="dir", tf_style="arrows", colors=["green", "red"])
+        .fmt_currency(columns=["open", "close"])
+        .cols_label(
+            open="Opening",
+            close="Closing",
+            dir=""
+        )
+    )
+    ```
+    """
+    # If colors is a string, convert it to a list
+    if isinstance(colors, str):
+        colors = [colors]
+
+    pf_format = partial(
+        fmt_tf_context,
+        data=self,
+        tf_style=tf_style,
+        pattern=pattern,
+        true_val=true_val,
+        false_val=false_val,
+        na_val=na_val,
+        colors=colors,
+    )
+
+    return fmt_by_context(self, pf_format=pf_format, columns=columns, rows=rows)
+
+
+def fmt_tf_context(
+    x: float,
+    data: GTData,
+    tf_style: str,
+    pattern: str,
+    true_val: str | None,
+    false_val: str | None,
+    na_val: str | None,
+    colors: list[str] | None,
+    context: str,
+) -> str:
+    # If `x` is not a boolean value, return it as is
+    if not isinstance(x, bool) and not is_na(data._tbl_data, x):
+        return x
+
+    # Obtain the list of `True`/`False` text values
+    tf_vals_list = _get_tf_vals(tf_style=tf_style)
```

Comment:
I would have this take true_val and false_val, and return the final `tf_vals_list`, so you don't have to wrangle it in the if/else blocks below. (they could default to None in `_get_tf_vals()` if useful)

Reply:
This recommended change has now been implemented.